package_nso_to_oc/xe/xe_system.py

```diff
@@ -864,6 +868,281 @@ def cleanup_server_access(config_leftover, group_access_type, access_type):
+                    if vrf_info.get(host_vrf):
+                        intf_vrf = vrf_info.get(host_vrf)
+                        break
+                if intf_ip_name_dict.get(intf_vrf):
+                    source_ip_vrf = intf_ip_name_dict.get(intf_vrf)
+            
+            temp_host = {"openconfig-system:host": f'{host}',
+                        "openconfig-system:config": get_host(host, source_ip_vrf, host_vrf),
+                        "openconfig-system:selectors": {
+                            "openconfig-system:selector": get_facility_severity(severity)
+                        }}
+            hosts_list.append(temp_host)
+            config_leftover["tailf-ned-cisco-ios:logging"]["host"]["ipv4-vrf"][index]["host"] = None
+            config_leftover["tailf-ned-cisco-ios:logging"]["host"]["ipv4-vrf"][index]["vrf"] = None
+    
+    if logging.get("facility"):
+        del config_leftover["tailf-ned-cisco-ios:logging"]["facility"]
+    if logging.get("trap"):
+        del config_leftover["tailf-ned-cisco-ios:logging"]["trap"]
+    cleanup_logging(config_leftover)
+    return hosts
+
+def get_severity(logging_severity):
+    # GET LOGGING SEVERITY
+    if logging_severity == "emergencies" or logging_severity == 0:
+        severity = "EMERGENCY"
+    elif logging_severity == "alerts" or logging_severity == 1:
+        severity = "ALERT"
+    elif logging_severity == "critical" or logging_severity == 2:
+        severity = "CRITICAL"
+    elif logging_severity == "errors" or logging_severity == 3:
+        severity = "ERROR"
+    elif logging_severity == "warnings" or logging_severity == 4:
+        severity = "WARNING"
+    elif logging_severity == "notifications" or logging_severity == 5:
+        severity = "NOTICE"
+    elif logging_severity == "informational" or logging_severity == 6:
+        severity = "INFORMATIONAL"
+    elif logging_severity == "debugging" or logging_severity == 7:
+        severity = "DEBUG"
+
+    return severity
+
+def get_host(host, source_ip, host_vrf):
+    # GET HOST, VRF AND SOURCE IP ADDRESS
+    temp_host = {"openconfig-system:host": f'{host}', 
+                    "openconfig-system:remote-port": 514,
+                    "openconfig-system:source-address": f'{source_ip}',
+                    "openconfig-system-ext:use-vrf": f'{host_vrf}'
+                }
+
+    return temp_host
+
+def get_facility_severity(severity):
+    # GET FACILITY AND SEVERITY
+    temp_fac_sev = [{"openconfig-system:facility": "SYSLOG", 
+                        "openconfig-system:severity": f'{severity}',
+                        "openconfig-system:config": {
+                        "openconfig-system:facility": "SYSLOG",
+                        "openconfig-system:severity": f'{severity}'
+                    }}]
+
+    return temp_fac_sev
+
+def cleanup_logging(config_leftover):
+    if len(config_leftover.get("tailf-ned-cisco-ios:logging", {}).get("source-interface", [])) < 1:
```

Comment:
Most times, you can avoid these type of empty ifs by checking for the opposite, like this

```suggestion
    if len(config_leftover.get("tailf-ned-cisco-ios:logging", {}).get("source-interface", [])) >= 1:
```